### ISSUE TITLE
Donchian Channel problem

### ISSUE THREAD

**Issue by AydinIst** [DaveSkender/Stock.Indicators#607]
Hi Dave.

I would like to talk about a situation I noticed while working with Donchian Channels.

I think that the Upper Band of the Donchian Channel and the High value of the bar closing should be equal.

And also the Lower Band of the Donchian Channel and the Low of the bar closing need to be equal.

But these values do not match. When I check these values on TradingView, there is no problem.

E.g;
on TradingView
GRTUSDTPERP - BINANCE - 15m - 10Nov2021 00:15 - High : **1.27848** and Donchinal Channel(20) Value : **1.27848**
Skender Indicator UpperBand Value : **1.25995**

Another example;
GRTUSDTPERP - BINANCE - 15m - 09Nov2021 20:00 - Low : **1.1900** and Donchinal Channel(20) Value : **1.1900**
Skender Indicator LowerBand Value : **1.19547**

Can you check?

**Comment by DaveSkender** [comment]
How confident are you that TradingView is correct? 😃 
I'll take a closer look.  Thanks for mentioning it.

If I remember this one correctly from when I was implementing, I noticed there was a broad misconception on this one, and it has probably been implemented differently (incorrectly) by several charting tools.  I'm often skeptical of TradingView because many of its indicators are untested community contributions, and prone to errors and "customizations" that differ from original formulas.

The basic gist of what I remember is that Richard Donchian used a crossover as a trade signal.  If the high/low is never allowed to cross the channel line, that could never happen.

I'll look back at my notes and double-check.

**Comment by DaveSkender** [comment]
As a reference, I'm using the same interpretation as StockCharts.com (a more reputable source):
https://school.stockcharts.com/doku.php?id=technical_indicators:price_channels

**Comment by DaveSkender** [comment]
Here's another reference:
https://corporatefinanceinstitute.com/resources/knowledge/trading-investing/donchian-channels/

![image](https://user-images.githubusercontent.com/8432125/141033448-6b3bb7a3-26f0-49b4-8753-ead6f9d6221f.png)

**Comment by AydinIst** [comment]
> How confident are you that TradingView is correct? 😃

Yes, I totally agree with you.

I also reviewed the sources you gave and now I checked again on the chart.

I detected this. If we go from the example above;

GRTUSDTPERP - BINANCE - 15m - 10Nov2021 - 00:15 - High : 1.27848 Donchian : 1.27848 (According to the TradingView :))
Skender value was 1.25995 (This value is previous bar's high value. It means 00:00 time's bar high)

As far as I understand, it is based on the high number of the previous bar when calculating via Skender. The current closing bar is ignored.
